src/main/scala/cromwell/engine/backend/CommandLineBackend.scala

```diff
@@ -0,0 +1,37 @@
+package cromwell.engine.backend
```

Comment:
Since each backend will need a few classes defined, Perhaps it's a good idea to make a `cromwell.engine.backend.local` package where this file lives (Maybe a better name is `LocalBackend`?  I don't care all that much...)

Then, I'd define a companion object here where you write your lookup function (a `String => WdlValue` function).  It'll look something like this:

```scala
def lookup_function(symbols: SymbolStore): String => WdlValue = {
  def lookup(identifier: String): WdlValue = {
    // use `symbols` to look up `identifier`
  }
  return lookup
}
```

Then perhaps move/rename the classes `EngineFunctions` and `TaskExecutionContext` into `cromwell.engine.backend.local`.  That gives a framework for implementing the WdlFunctions.

Finally, to evaluate a taskOutput, the code looks something like this:

```scala
taskOutput.evaluate(
  CommandLineBackend.lookup_function(symbolStore),
  new EngineFunctions(new TaskContext(stdoutFile, stderrFile))
)
```

This way, every time the expression evaluator needs to resolve an identifier, it calls your lookup function.  Every time it needs to evaluate a function, it calls the corresponding function in EngineFunctions!